db/migrate/20240219100348_add_administrative_number_to_schoolings.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class AddAdministrativeNumberToSchoolings < ActiveRecord::Migration[7.1]
+  def migrate_single_attributive_decision!
+    Student
+      .joins(:schoolings)
+      .merge(Schooling.with_attributive_decisions)
+      .group(:id)
+      .having("count(schoolings.id) < 2")
+      .find_each do |student|
+      student.schoolings.first.update!(administrative_number: student.asp_file_reference)
```

Comment:
T'es pas sûr de target la bonne schooling là, avec ton `.first`.

Le mieux serait que ta requête avant le `find_each` sorte des schoolings, et que tu itères sur les schoolings que t'as isolées.